### ISSUE TITLE
CutOut should support fill_mode="mirror"

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#336]
This is actually a great "research" project in that this should strictly be better than using noise and black pixels.

**Comment by innat** [comment]
Can you add some references, sounds interesting.

**Comment by LukeWood** [comment]
There are references out there for mirroring in random translation and rotation, but not random erasing or cut out.   The logic *should* in theory extend to cutout, but I do not have definitive proof.  That's why this is also somewhat of a research effort.  Happy to collaborate with anyone on benchmarks if they are interested...

**Comment by bhack** [comment]
Are you talking about (https://arxiv.org/pdf/1708.04552v2.pdf):

![immagine](https://user-images.githubusercontent.com/1710528/164320214-b72a5a87-ab0c-469f-958a-b4ffc4781f00.png)

 or something else?

**Comment by bhack** [comment]
We have some experimental results with patch gaussian as an alternative "fill_mode" for cutout:

https://arxiv.org/abs/1906.02611

**Comment by innat** [comment]
The `fill_mode="mirror"` sounds like a `fill_mode="reflect"`.

**Comment by bhack** [comment]
The patch gaussian has a good citations threshod (>100).

There is a TF implementation at page 15.

I don't know if It Is better to spent time on something that has already some experimental results already.

**Comment by bhack** [comment]
https://github.com/tensorflow/models/blob/master/research/object_detection/core/preprocessor.py#L2793

**Comment by innat** [comment]
@bhack Thanks for sharing this information, very useful. It's definitely an effective option.

> Cutout, which improves clean accuracy but not robustness, and additive Gaussian noise, which improves robustness but hurts accuracy. To overcome this trade-off, we introduce Patch Gaussian, a simple augmentation scheme that adds noise to randomly selected patches in an input image.

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new KerasHub package, which supports image, text, and multi-modal models. Please read https://github.com/keras-team/keras-hub/issues/1831. KerasHub will support all the core functionality of KerasCV.

KerasHub can be installed with !pip install -U keras-hub. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).

**Comment by innat** [comment]
@sachinprasadhs 
> If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).

Firstly, identifying which feature is required or missing can be done effectively, either by practitioners or the Keras team. Tickets in `keras-cv` were created only when such features were **NOT** available or when a specific need was encountered. Commenting after a significant delay and requesting the recreation of the same issue is neither efficient nor appropriate. Kindly move this ticket (and any other similar tickets with the same context) from `keras-cv` to `keras-hub`.
